src/xdocs/config_coding.xml

```diff
@@ -36,16 +36,21 @@ int[] a = new int[]
         </source>
 
         <p>
-          The check allows leaving out the comma at the end if both the left and right curly brackets
-          are on the same line.
+          The check allows leaving out the comma at the end if either left
+          or right curly braces are on the same line as the last element of the array.
         </p>
         <source>
 return new int[] { 0 };
+return new int[] { 0
+        };
+return new int[] {
+        0 };
         </source>
 
         <p>
           Rationale: Putting this comma in makes it easier to change the order
-          of the elements or add new elements on the end.
+          of the elements or add new elements on the end. Main benefit of a trailing
+          comma is that when you add new entry to an array, no surrounding lines are changed.
```

Comment:
I still think that 4 examples that mkordas provided should be there (last two cases is the most problematic to understand).
@rnveach , please share your opinion.